tests/v3_api/test_project_quota.py

```diff
@@ -0,0 +1,162 @@
+                      "requestsCpu": "2000m"}}
+
+
+def ns_quota():
+    return {"limit": {"pods": "10",
+                      "requestsCpu": "500m"}}
+
+
+def test_create_project_quota():
+    # Create Project Resource Quota and verify quota is created
+    # successfully. Verify namespacedefault resource quota is set
+
+    cluster = namespace["cluster"]
+    client = get_admin_client()
+    c_client = namespace["c_client"]
+
+    quota = default_project_quota()
+    nsquota = ns_default_quota()
+    proj = client.create_project(name='test-' + random_str(),
+                                 clusterId=cluster.id,
+                                 resourceQuota=quota,
+                                 namespaceDefaultResourceQuota=nsquota)
+    proj = client.wait_success(proj)
+
+    assert proj.resourceQuota is not None
+    assert proj.resourceQuota.limit.pods == '20'
+    assert proj.resourceQuota.limit.requestsCpu == '2000m'
+    assert proj.namespaceDefaultResourceQuota is not None
+    assert proj.namespaceDefaultResourceQuota.limit.pods == '5'
+
+    ns = create_ns(c_client, cluster, proj)
+    print(ns)
+
+    assert ns is not None
+    assert ns.resourceQuota is not None
+    assert ns.resourceQuota.limit.pods == "5"
+    assert ns.resourceQuota.limit.requestsCpu == '500m'
+
+    validate_resoucequota_thru_kubectl(ns)
+
+
+def test_resource_quota_create_namespace_with_ns_quota():
+
+    # Create project quota and namspaces and verify
+    # namespace creation is allowed within the quota
+
+    cluster = namespace["cluster"]
+    client = get_admin_client()
+    c_client = namespace["c_client"]
+
+    quota = default_project_quota()
+    proj = client.create_project(name='test-' + random_str(),
+                                 clusterId=cluster.id,
+                                 resourceQuota=quota,
+                                 namespaceDefaultResourceQuota=quota)
+    proj = client.wait_success(proj)
+
+    assert proj.resourceQuota is not None
+
+    # Create a namespace with namespace resource quota
+    ns_name = random_str()
+    ns = c_client.create_namespace(name=ns_name,
+                                   projectId=proj.id,
+                                   resourceQuota=ns_quota())
+    ns = c_client.wait_success(ns)
+
+    assert ns is not None
+    assert ns.resourceQuota is not None
+    assert ns.resourceQuota.limit.pods == "10"
+    assert ns.resourceQuota.limit.requestsCpu == '500m'
+
+    validate_resoucequota_thru_kubectl(ns)
+
+    # Create another namespace with quota and it should succeed
+
+    ns1 = c_client.create_namespace(name=random_str(),
+                                    projectId=proj.id,
+                                    resourceQuota=ns_quota())
+    ns1 = c_client.wait_success(ns1)
+    print(ns1)
+
+    assert ns1 is not None
+    assert ns1.resourceQuota is not None
+    assert ns1.resourceQuota.limit.pods == "10"
+    assert ns1.resourceQuota.limit.requestsCpu == '500m'
```

Comment:
use ns_quota dictionary lookup instead of hardcoded value

Reply:
Done